assets/src/edit-story/components/canvas/canvasUploadDropTarget.js

```diff
@@ -59,16 +60,28 @@ function CanvasUploadDropTarget({ children }) {
   );
 
   const onUploadedFile = useCallback(
-    ({ resource, element }) => {
+    async ({ resource, element }) => {
+      const keysToUpdate = objectPick(resource, [
+        'src',
+        'width',
+        'height',
+        'length',
+        'lengthFormatted',
+        'videoId',
+      ]);
+      const updatedResource = {
+        ...element.resource,
+        ...keysToUpdate,
+      };
       updateElementById({
         elementId: element.id,
         properties: {
-          resource,
-          type: element.resource.type,
+          resource: updatedResource,
         },
       });
+      await uploadVideoPoster(resource.videoId, resource.src);
```

Comment:
So on upload trigger the videoPoster generation? This make sense, as we now have a processing queue, that should mean there are not processing.

Reply:
I would still add a check for upload ( attachment ) type.